public/friendlycode/js/fc/filesystem-sync.js

```diff
@@ -0,0 +1,104 @@
+define(["jquery"], function($) {
+  "use strict";
+
+  var FileSystemSync = {};
+
+  function pushFileChange(url, csrfToken, fs, path) {
+    var options = {
+      contentType: "application/json",
+      headers: {
+        "X-Csrf-Token": csrfToken
+      },
+      type: "PUT",
+      url: url
+    };
+
+    function send() {
+      var request = $.ajax(options);
+      request.done(function() {
+        if(request.readyState !== 4) {
```

Comment:
You don't need this, since `.done` callbacks are executed when the parent promise resolves. In other words, this will only execute when the request is finished.